crates/nargo/src/cli/compile_cmd.rs

```diff
@@ -45,7 +45,10 @@ pub(crate) fn run(mut args: CompileCommand, config: NargoConfig) -> Result<(), C
         }
         Ok(())
     } else {
-        let main = driver.main_function();
+        let main = match driver.main_function() {
+            Ok(m) => m,
+            Err(_) => return Err(CliError::CompilationError),
+        };
```

Comment:
```suggestion
        let main = driver.main_function().map_err(|_| CliError::CompilationError)?;
```
We can also do this in the places we're doing `is_err` if statements.